basic/src/test/java/org/jboss/hal/testsuite/test/configuration/undertow/UndertowTestCaseAbstract.java

```diff
@@ -50,47 +43,65 @@
 
     @BeforeClass
     public static void mainSetUp() throws IOException, CommandFailedException, TimeoutException, InterruptedException {
-        dispatcher = new Dispatcher();
-        verifier = new ResourceVerifier(dispatcher);
-        operations = new UndertowOperations(ManagementClientProvider.createOnlineManagementClient());
-        WORKER_VALUE_VALID = operations.createWorker();
-        BUFFER_POOL_VALUE_VALID = operations.createBufferPool();
-        SOCKET_BINDING_VALUE_VALID = operations.createSocketBinding();
+        client = ManagementClientProvider.createOnlineManagementClient();
+        operations = new Operations(client);
+        administration = new Administration(client);
+        undertowOps = new UndertowOperations(client);
     }
 
     @AfterClass
-    public static void mainTearDown() throws InterruptedException, IOException, TimeoutException {
-        operations.removeWorker(WORKER_VALUE_VALID);
-        operations.removeBufferPool(BUFFER_POOL_VALUE_VALID);
-        dispatcher.close();
+    public static void mainTearDown() throws InterruptedException, IOException, TimeoutException, OperationException, CommandFailedException {
+        try {
+            undertowOps.close();
+            administration.reloadIfRequired();
+            administration.restartIfRequired();
+        } finally {
+            client.close();
+        }
+    }
+
+    protected void editTextAndVerify(Address address, String attributeName, String value) throws Exception {
+        page.getConfigFragment().editTextAndSave(attributeName, value);
+        administration.reloadIfRequired();
+        new ResourceVerifier(address, client).verifyAttribute(attributeName, value);
     }
 
-    protected void editTextAndVerify(ResourceAddress address, String identifier, String attributeName, String value) throws IOException, InterruptedException, TimeoutException {
-        page.getConfigFragment().editTextAndSave(identifier, value);
-        operations.reloadIfRequiredAndWaitForRunning();
-        verifier.verifyAttribute(address, attributeName, value);
+    protected void editTextAndVerify(Address address, String attributeName, int value) throws Exception {
```

Comment:
I'm not sure if in 100% the UI `identifier` should be same as model `attributeName`. See `SMAP` above. May be you should add method with one more attribute `identifier` for cases when `identifier` is different from `attributeName` .